src/components/ThreadItem/utils.ts

```diff
@@ -0,0 +1,103 @@
+export function parseThreadCover(blocks: any[]) {
+    let block = blocks.find(b => b.type === "image" || b.type === "embed");
+    if (block) {
+        if (block.type === "image") {
+            return [block.data.file.url, "image"];
+        }
+
+        if (block.type === "embed") {
+            return [getVideoCover(block.data.source), "video"];
+        }
+    }
+    return [null, null];
+}
+
+export const getEmbedUrl = (href: string) => {
+    let [source, embedId] = getEmbedId(href);
+    if (source === "youtube") {
+        return createYoutubeEmbedUrl(embedId);
+    }
+    if (source === "vimeo") {
+        return createVimeoEmbedUrl(embedId)
+    }
+}
+
+export const getVideoCover = (href: string) => {
+    let [source, embedId] = getEmbedId(href);
+    if (source === "youtube") {
+        return createYoutubeCover(embedId);
+    }
+    if (source === "vimeo") {
+        return createVimeoCover(embedId)
+    }
+}
+
+export const getEmbedSource = (href: string) => {
+    let isYoutubeWithWatch = new RegExp(/^(http:\/\/|https:\/\/)(youtu\.be|www\.youtube\.com|youtube\.com)(\/watch)([^\s]+)/, "gmi");
+    let isYoutube = new RegExp(/^(http:\/\/|https:\/\/)(youtu\.be|www\.youtube\.com|youtube\.com)([\w\/-]+)([^\s]+)/, "gmi");
+    let isVimeo = new RegExp(/^(http:\/\/|https:\/\/)(player\.vimeo\.com|vimeo\.com)([\w\/-]+)([^\s]+)/, "gmi");
+
+    if (isYoutubeWithWatch.test(href)) {
+        return "youtube";
+    }
+    if (isYoutube.test(href)) {
+        return "youtube";
+    }
```

Comment:
```suggestion
    let isYoutube = new RegExp(/^(http:\/\/|https:\/\/)(youtu\.be|www\.youtube\.com|youtube\.com)(\/watch)?([\w\/-]+)?([^\s]+)/, "gmi");
   
    let isVimeo = new RegExp(/^(http:\/\/|https:\/\/)(player\.vimeo\.com|vimeo\.com)([\w\/-]+)([^\s]+)/, "gmi");

    if (isYoutube.test(href)) {
        return "youtube";
    }
```

To be tested, not 100% sure.

Reply:
To DRY this expressions in this file